tests/green_mode/green_modeTest.py

```diff
@@ -356,5 +374,150 @@ def test_bear_test_fun_2(self):
                               test_non_op_results[0][TestLocalBear])
         self.assertCountEqual(unified_results, [None, None])
 
+    def test_write_coafile(self):
+        from pyprint.ConsolePrinter import ConsolePrinter
+        printer = ConsolePrinter()
+        input_ = [{TestLocalBear: [{'filename': 'a.py', 'some_setting': 3},
+                                   {'filename': 'b.py', 'some_setting': 3},
+                                   {'filename': 'c.py', 'some_setting': 4},
+                                   {'filename': 'd.py',
+                                    'some_other_setting': 'x'}]},
+                  None]
+        data_struct = generate_data_struct_for_sections(input_)
+        test_data_struct = {TestLocalBear: [[{'filename': ['a.py', 'b.py'],
+                                              'some_setting': 3},
+                                             {'filename': ['c.py'],
+                                              'some_setting': 4}],
+                                            [{'some_other_setting': 'x',
+                                              'filename': ['d.py']}]]}
+        self.assertEqual(data_struct, test_data_struct)
+        project_files = ['a.py', 'b.py', 'c.py', 'd.py']
+        coafile = '.coafile.green'
+        full_path = str(Path(__file__).parent.parent.parent)
+        full_path_coafile = str(Path(__file__).parent.parent.parent / coafile)
+        with patch('os.walk') as mockwalk:
+            mockwalk.return_val = mockwalk.return_value = [
+                ('', (), ('a.py', 'b.py', 'c.py', 'd.py')), ]
+            generate_green_mode_sections(data_struct, full_path,
+                                         project_files, ['x'],
+                                         printer)
+        contents = ""
+
+        with open(full_path_coafile) as f:
+            for line in f.readlines():
+                contents += line
+        full_path_glob = escape(full_path + os.sep + '**', '\\')
+        test_contents = dedent("""
+                            [all]
```

Comment:
This must be done properly.
the textual content must be indented only four spaces from the beginning of the previous line.  i.e. under `_` in `test_contents`

Reply:
Still not done. Read what I wrote.  Carefully!

Read the docs of textwrap.dedent to see how it should be used.